src/ucp/api/ucp.h

```diff
@@ -3135,6 +3108,186 @@ ucs_status_ptr_t ucp_worker_flush_nb(ucp_worker_h worker, unsigned flags,
                                      ucp_send_callback_t cb);
 
 
+/**
+ * @brief UCP active message rendezvous parameters field mask.
+ *
+ * The enumeration allows specifying which fields in @ref ucp_params_t are
+ * present. It is used to enable backward compatibility support.
+ */
+enum ucp_am_params_field {
+  UCP_AM_PARAMS_FIELD_FLAGS      = UCS_BIT(0), /* Flags field */
+  UCP_AM_PARAMS_FIELD_IOVEC_SIZE = UCS_BIT(1)  /* Max size of the */
+                                               /* iovec returned by */
+                                               /* the initial AM */
+};
+
+/**
+ * @brief Tuning parameters for the UCP active message.
+ *
+ * The structure defines the parameters that are used for the
+ * UCP active message function
+ */
+typedef struct ucp_am_params {
+  uint64_t field_mask;
+  uint64_t flags;
+
+  size_t   iovec_size; /* Number of elements that the ucp library should */
+                       /* allocate in the 'iovec 'for the callback to */
+                       /* fill in. */
+} ucp_am_params_t;
+
+/**
+ * @ingroup UCP_COMM
+ * @brief Completion callback for data arrivals.
+ *
+ * This callback routine is invoked whenever the @ref ucp_tag_send_nb
+ * The ucp library will drive a function on the arrival of each fragment
+ * of data from the initiator.
+ *
+ * @param [out} target    Where the data should be placed. NULL if no copying
+ *                        needed.
+ * @param [in}  source    Where the data currently is
+ * @param [in]  bytes     How much data has just arrived
+ * @param [in]  cookie    Data cookie from the ucp_am_recv structure
+ */
+typedef void (*ucp_am_data_function_t)(void *target, void *source,
+                                       size_t bytes, void *cookie);
+
+/**
+ * @ingroup UCP_COMM
+ * @brief Completion callback for transfer from initiator complete.
+ *
+ * The ucp library will drive a function when the transfer from the
+ * initiator is complete.
+ *
+ * @param [in}  arg             Request structure
+ * @param [in}  cookie          Cookie from the ucp_am_recv structure
+ * @param [in]  iovec           Description of the received data
+ * @param [in]  iovec_length    Number of elements in iovec
+ */
+typedef ucs_status_t (*ucp_am_local_function_t)(void *arg, void *cookie,
+                                                ucp_dt_iov_t *iovec,
+											 size_t iovec_length);
+/**
+ * @brief UCP active message rendezvous redeive parameters field mask.
+ *
+ * The enumeration allows specifying which fields in @ref ucp_am_recv are
+ * present. It is used to enable backward compatibility support.
+ */
+enum ucp_am_params_recv {
+  UCP_AM_PARAMS_RECV_FIELD_FLAGS      = UCS_BIT(0) /* Flags field */
+};
+/*
```

Comment:
This is not doxygen formatted. Everything should be doxygen formated in order to generate nice document

Reply:
Can you point me to an example that I should follow ?

Reply:
This a good reference http://www.doxygen.nl/manual/docblocks.html
For the single liner like above, you want to do ```/**< xxxx */```

Reply:
Also, there is bunch of similar declarations cup.h itself.